### ISSUE TITLE
Simpleauth dont work with long username or passwords

### ISSUE THREAD

**Issue by mariocesar** [unbit/uwsgi#161]
Using: Starting uWSGI 1.9-dev (64bit) compiled version of the revision https://github.com/unbit/uwsgi/commit/61a1ca4ca4445b7985fb38d5278cf3aabe22d235

I'm using the last Have this uwsgi setup.

``` ini
[uwsgi]
master       = true
workers      = 2
static_root  = %d
route        = ^/ basicauth:Realm,admin:admin
route        = ^/ basicauth:Realm,

route        = ^/robots.txt$ static:%(static_root)robots.txt
route        = ^/humans.txt$ static:%(static_root)humans.txt
```

On the same directory I have the the robots.txt and humans.txt files, everything works fine:

```
$ wsgi --ini uwsgi.ini --http :8000
$ curl -u admin:admin http://127.0.0.1:8000/humans.txt
Hello world!
```

The Issue, if I use a longer password it dont work.  

```
route        = ^/ basicauth:Realm,admin:een0ai5PUsh6oeNg
route        = ^/ basicauth:Realm,user2:een0ai5PUsh6oeNg
```

Test:

```
$ curl -u admin:een0ai5PUsh6oeNg http://127.0.0.1:8000/humans.txt
Unauthorized
$ curl -u user:een0ai5PUsh6oeNg http://127.0.0.1:8000/humans.txt
Unauthorized
```

TL;DR if both the password and the username are longer than 15 chars, simpleauth don't work.
